main/solution/backend/config/infra/cloudformation.yml

```diff
@@ -577,9 +586,18 @@ Resources:
               - !Sub 'arn:aws:iam::${AWS::AccountId}:role/*-xacc-env-mgmt'
           - Effect: Allow
             Action:
-              - logs:CreateLog*
+              - logs:CreateLogGroup
+              - logs:CreateLogStream
               - logs:PutLogEvents
-            Resource: !Sub 'arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:aws/lambda/${self:custom.settings.awsRegionShortName}-${self:custom.settings.solutionName}-backend-${self:custom.settings.envName}*'
+            Resource: 
+              - !Sub 'arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/${self:custom.settings.awsRegionShortName}-${self:custom.settings.solutionName}-backend-${self:custom.settings.envName}-workflowLoopRunner:*'
+              - !Sub 'arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/${self:custom.settings.awsRegionShortName}-${self:custom.settings.solutionName}-backend-${self:custom.settings.envName}-workflowLoopRunner:log-stream:*'
+          - Effect: Allow
+            Action:
+              - xray:PutTraceSegments
+              - xray:PutTelemetryRecords
+            Resource:
+              - '*'
```

Comment:
Can the resource here be more specific? Can we add the namespace here at least?

Reply:
Base on the documentation, it looks like we can't narrow the scope.
https://docs.aws.amazon.com/xray/latest/devguide/security_iam_id-based-policy-examples.html

```
For actions that don't support resource-level permissions, you must use "*" as the resource.

The following X-Ray actions support resource-level permissions:
    CreateGroup
    GetGroup
    UpdateGroup
    DeleteGroup
    CreateSamplingRule
    UpdateSamplingRule
    DeleteSamplingRule
```

`PutTraceSegments` and `PutTelemetryRecords` are not in that list.

Reply:
Got it thanks